archai/nlp/models/hf_gpt2/model_hf_gpt2.py

```diff
@@ -32,6 +32,9 @@ def __init__(self, **kwargs) -> None:
 
         if self.config.tie_weight:
             self.model.tie_weights()
+            # Set the shared attribute so that Opacus knows that
+            # hte gradients must be summed and are not a new batch
+            self.model.lm_head.shared = True
```

Comment:
Is this a special attribute Opacus specifically looks for? Also, could you please add comment with reference to more details?

Reply:
No. It is a attribute that my patch (Opacus changes) will look for. Will add more details.